utils/ixdtf/src/parsers/records.rs

```diff
@@ -0,0 +1,201 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Parse records for `ixdtf`'s parsers to target.
+
+use alloc::vec::Vec;
+
+/// An `IsoParseRecord` is an intermediary record returned by ISO parsing functions.
+///
+/// `IsoParseRecord` is converted into the ISO AST Nodes.
+#[non_exhaustive]
+#[derive(Default, Debug, PartialEq)]
+pub struct IsoParseRecord<'a> {
+    /// Parsed Date Record
+    pub date: Option<DateRecord>,
+    /// Parsed Time
+    pub time: Option<TimeRecord>,
+    /// Parsed UtcOffset
+    pub offset: Option<UTCOffset>,
+    /// Parsed `TimeZone` data (UTCOffset | IANA name)
+    pub tz: Option<TimeZone<'a>>,
+    /// The parsed calendar value.
+    pub calendar: Option<&'a str>,
+    /// Annotations
+    pub annotations: Vec<Annotation<'a>>,
+}
+
+#[non_exhaustive]
+#[derive(Debug, Clone, PartialEq)]
+pub struct Annotation<'a> {
+    pub critical: bool,
+    pub key: &'a str,
+    pub value: &'a str,
+}
+
+#[non_exhaustive]
```

Comment:
Suggestion: `DateRecord` and `TimeRecord` can probably be made exhaustive since by specification these are the only fields that could appear. Probably good to keep `IsoParseRecord` as non-exhaustive since in principle we could pull out more of the annotations to fields. Not sure about `Annotation`.

Reply:
I did remove the `#[non_exhaustive]` from the struct. 

As a general side note / follow up, even if the records are exhaustive, would there be any benefit in keeping them as `#[non_exhaustive]` for semver considerations and to also prevent the structs from being constructed by a `StructExpression`? Although, I don't really see any reason any `ixdtf` user would have a need for that functionality.

Reply:
Exhaustive structs are a bit more useful to users because you can unpack them / match on them without needing a catchall code path, but non-exhaustive lets the ICU4X team add things without breaking semver. We have this lint primarily so that we make an explicit choice for each thing we export